torch/csrc/distributed/c10d/Ops.cpp

```diff
@@ -31,6 +31,17 @@ c10::intrusive_ptr<ProcessGroup::Work> allreduce_(
           std::chrono::milliseconds(timeout)});
 }
 
+c10::intrusive_ptr<ProcessGroup::Work> allgather_(
+    const std::vector<std::vector<at::Tensor>>& output_tensors,
+    const std::vector<at::Tensor>& input_tensors,
+    const c10::intrusive_ptr<ProcessGroup>& process_group,
+    int64_t timeout) {
+  return process_group->allgather(
+      const_cast<std::vector<std::vector<at::Tensor>>&>(output_tensors),
```

Comment:
We should be able to modify allgather to take `const std::vector` as well if necessary. Shall we create an issue to track that?

Reply:
Yea, I agreed. PG's should accept const vectors& instead of vectors& given what we do is to update the content of the values within the vectors and not the vectors itself. Let me create an issue. #80241.